### PR TITLE
Implement CustomNameData

### PR DESCRIPTION
**Pull request by kashike**
[API](https://github.com/SpongePowered/SpongeAPI/pull/1013) | [**Common**](https://github.com/SpongePowered/SpongeCommon/pull/403)

### CODE REVIEW

**Comment by Aaron1011** [comment]
@kashike: You need to take into account the fact that `Humanoid`s always have a display name visible.

File: src/main/java/org/spongepowered/common/data/processor/dual/entity/CustomNameVisibleDualProcessor.java [hunk 1]
```diff
@@ -60,11 +65,27 @@ protected boolean set(Entity container, Boolean value) {
...
     }
+
+    @Override
+    protected boolean supports(Entity container) {
+        return !(container instanceof Humanoid);
+    }
+
+    @Override
+    public boolean supports(DataHolder container) {
+        return !(container instanceof Humanoid);
```
Comment (Aaron1011) [inline src/main/java/org/spongepowered/common/data/processor/dual/entity/CustomNameVisibleDualProcessor.java]: Is it necessary to override this method as well?

File: src/main/java/org/spongepowered/common/data/processor/data/DisplayNameDataProcessor.java [hunk 2]
```diff
@@ -40,156 +38,141 @@
...
-        final Text displayName = TextSerializers.JSON.deserialize(json);
-        final boolean shows = DataUtil.getData(container, Keys.SHOWS_DISPLAY_NAME);
-        return Optional.of(displayNameData.set(Keys.DISPLAY_NAME, displayName).set(Keys.SHOWS_DISPLAY_NAME, shows));
+        return Optional.of(displayNameData.set(Keys.DISPLAY_NAME, TextSerializers.JSON.deserialize(json)));
     }
 
     @Override
-    public DataTransactionResult set(DataHolder dataHolder, DisplayNameData manipulator, MergeFunction function) {
+    public DataTransactionResult set(DataHolder holder, DisplayNameData manipulator, MergeFunction function) {
         return DataTransactionResult.failResult(manipulator.getValues());
```
Comment (Aaron1011) [inline src/main/java/org/spongepowered/common/data/processor/data/DisplayNameDataProcessor.java]: This, and and several other methods in this class, don't need to be overriden anymore.

**Comment by Aaron1011** [comment]
@kashike: Could you rebase this against master?

File: src/main/java/org/spongepowered/common/data/processor/data/DisplayNameDataProcessor.java (hunk 2 above)
Comment (kashike) [inline src/main/java/org/spongepowered/common/data/processor/data/DisplayNameDataProcessor.java]: Nope, need to.

**Comment by gabizou** [comment]
Closing as this was merged into [bleeding](https://github.com/SpongePowered/SpongeCommon/commit/526a3adcb723ce4c15dc0e3943519a95f950ed4e).

**Pull request closed without merging**